lib/saml/settings_service.rb

```diff
@@ -31,9 +34,9 @@ def settings
     end
 
     ## Makes an external web call to get IDP metadata and populates SETTINGS
-    def fetch_idp_metadata
+    def create_settings_from_metadata
       parser = OneLogin::RubySaml::IdpMetadataParser.new
-      parser.parse_remote(SAML_CONFIG['metadata_url'], true, settings: settings)
+      parser.parse(self.class.metadata, settings: settings)
```

Comment:
Skipping `parse_remote` in favor of `parse` means you skip cert validation.  From the ruby-saml source code:
```ruby
def parse_remote(url, validate_cert = true, options = {})
  idp_metadata = get_idp_metadata(url, validate_cert)
  parse(idp_metadata, options)
end
```
We _must_ validate the cert when fetching metadata.